python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py

```diff
@@ -0,0 +1,262 @@
+                                         thread_num=thread_num,
+                                         # remove output of openvino
+                                         logging=logging)
+        else:
+            # quantize
+            ort_method: str = self.method
+            acce_model = model.quantize(precision=self.get_precision(),
+                                        accelerator=accelerator,
+                                        calib_dataset=training_data,
+                                        method=ort_method,
+                                        thread_num=thread_num,
+                                        sample_size=sample_size_for_pot,
+                                        # remove output of openvino
+                                        logging=logging)
+        return acce_model
+
+
+class InferenceOptimizer(BaseInferenceOptimizer):
+
+    # acceleration method combinations, developers may want to register some new
+    # combinations here
+    ALL_INFERENCE_ACCELERATION_METHOD: Dict = \
+        {  # type: ignore
+            "original": TFAccelerationOption(),
+            "int8": TFAccelerationOption(inc=True),
+            "openvino_fp32": TFAccelerationOption(openvino=True),
+            "openvino_int8": TFAccelerationOption(openvino=True, pot=True),
+            "onnxruntime_fp32": TFAccelerationOption(onnxruntime=True),
+            "onnxruntime_int8_qlinear": TFAccelerationOption(onnxruntime=True, inc=True,
+                                                             method="qlinear"),
+            "onnxruntime_int8_integer": TFAccelerationOption(onnxruntime=True, inc=True,
+                                                             method="integer"),
+        }  # type: ignore
+
+    def optimize(self, model: Model,
+                 training_data: Dataset,
+                 validation_data: Optional[Dataset] = None,
+                 batch_size: int = 1,
+                 metric: Optional[Metric] = None,
+                 direction: str = "max",
+                 thread_num: Optional[int] = None,
+                 logging: bool = False,
+                 latency_sample_num: int = 100,
+                 includes: Optional[List[str]] = None,
+                 excludes: Optional[List[str]] = None) -> None:
+        '''
+        This function will give all available inference acceleration methods a try
+        and record the latency, accuracy and model instance inside the Optimizer for
+        future usage. All model instance is setting to eval mode.
+
+        The available methods are "original", "openvino_fp32", "onnxruntime_fp32", "int8".
+
+        :param model: A keras.Model to be optimized
+        :param training_data: An unbatched tf.data.Dataset object for for training dataset.
+               Users should be careful with this parameter since this dataloader
+               might be exposed to the model, which causing data leak.
```

Comment:
I am not quite sure I understand the document. Maybe explaining more? (For example, how to be careful?)

Reply:
Have update this note.